src/main/java/org/junit/runner/Result.java

```diff
@@ -156,13 +166,15 @@ public RunListener createListener() {
         private static final long serialVersionUID = 1L;
         private final AtomicInteger fCount;
         private final AtomicInteger fIgnoreCount;
+        private final AtomicInteger assumptionFailureCount;
```

Comment:
I'm reasonably sure that if you add this, you'll be breaking serialization. It's possible that new code will be able to read objects serialized with old code (the tests in `ResultTests.java` check for that) but code linked against 4.12 won't be able to read `Result` objects serialized by versions after 4.12. That _may_ break some tests or IDEs.

Could we instead not add this field, and have the `Result(SerializedForm)` constructor set a field to indicate that the assumption failure count cannot be determined?